numpy/linalg/_linalg.py

```diff
@@ -134,6 +135,8 @@ def isComplexType(t):
                       cdouble: cdouble}
 
 def _realType(t, default=double):
+    if finfo(clongdouble).bits == finfo(cdouble).bits:
```

Comment:
From the sidelines (not knowing what this function is doing because there's no docstring):

- should there be a corresponding entry in the `_complexType` function?
- all the values in the `_real_types_map` dictionary are real types, so should the following line be `_real_types_map[clongdouble] = double`?
- should there be an entry `longdouble: double`? Perhaps not by default, but with a logical test like you've just added.

Reply:
Probably right on the need to map to `double` instead of `cdouble`, suggestions for regression tests that can tell the difference for those things may be good though.

Reply:
I don't precisely know what the function is for, if I knew I could help :-(.

Reply:
Would this work for the test?

```
@pytest.mark.skipif(np.finfo(np.clongdouble).bits != np.finfo(np.cdouble).bits)
def test_linalg_mapping():
    assert _realType(np.clongdouble) is np.double 
    assert _realType(np.longdouble) is np.double
    assert _complexType(np.clongdouble) is np.cdouble
    assert _complexType(np.longdouble) is np.cdouble
```

Reply:
I can try it, I guess I was thinking more of a way to hit the codepaths via public API in a way that is sensitive